### DISCUSSION TITLE
How to Override ID type.

### DISCUSSION THREAD

**Discussion by fernandortec** [geldata/gel#7496] [Q&A]
I tried overriding the ID to be an auto incrementing or an ULID, but I can't. I looked at the docs and there's no info on how to do it.
![image](https://github.com/edgedb/edgedb/assets/62437672/cbc99f69-f1a9-46f9-b123-c7a862c9c31c)

**Comment by winnerlein** [comment]
Primary ids are always UUIDs in EdgeDB, you cannot change the type. I think currently it's using a variant of UUIDv1, which is timebased, somewhat similar to ULID, and will switch to UUIDv7 which is very similar to ULID (time based prefix, random suffix, strictly monotonic).

It uses the hex-with-hyphens format when outputting a UUID, but you could reformat it to base32 in your application code, if that's a part of ULID you care about.

You can also add an additional `external-id` property with `constraint exclusive` which contains an auto-incrementing id, or a ULID, or whatever you want.
